### PR TITLE
Add regression test for #98973

### PR DESCRIPTION
**Pull request by blasten**
This test exercises Choreographer#doFrame during application startup in debug, profile and release modes.

Trace: https://ui.perfetto.dev/#!/?s=da6628c3a92456ae8fa3f345d0186e781da77e90fc8a64d073e9fee11d1e65

Regression test for #98973

### CODE REVIEW

File: dev/devicelab/lib/tasks/android_choreographer_do_frame_test.dart [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+import 'package:flutter/services.dart';
+
+Future<void> main() async {
+  WidgetsFlutterBinding.ensureInitialized();
+
+  print('${kSentinelStr[0]}');
+  await SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersiveSticky);
+
+  print('${kSentinelStr[1]}');
+  await SystemChrome.setEnabledSystemUIMode(SystemUiMode.immersiveSticky);
```
Comment (jason-simmons) [inline dev/devicelab/lib/tasks/android_choreographer_do_frame_test.dart:68]: What does the second call to `setEnabledSystemUIMode(immersiveSticky)` do?

Comment (blasten) [reply dev/devicelab/lib/tasks/android_choreographer_do_frame_test.dart:68]: it should do nothing, but this future won't resolve if the main thread is blocked. I can add a comment to clarify.

File: .ci.yaml [hunk 2]
```diff
@@ -4323,3 +4323,14 @@ targets:
     properties:
       tags: >
           ["framework","hostonly","shard"]
+
+  - name: Mac_android android_choreographer_do_frame_test
```
Comment (keyonghan) [inline .ci.yaml:4327]: An entry is needed in [TESTOWNERS](https://github.com/flutter/flutter/blob/master/TESTOWNERS) to unblock the `Linux test_ownership` failure.

**Review by jason-simmons — APPROVED** [review]

File: .ci.yaml (hunk 2 above)
Comment (blasten) [reply .ci.yaml]: Done

**Review by keyonghan — APPROVED** [review]
LGTM

**Pull request merged**